app/models/structure.rb

```diff
@@ -1,18 +1,54 @@
+# == Schema Information
+#
+# Table name: structures
+#
+#  id             :bigint(8)        not null, primary key
+#  attempts       :integer          default(0)
+#  description    :text
+#  name           :string
+#  riddle         :integer
+#  structure_type :integer
+#  created_at     :datetime         not null
+#  updated_at     :datetime         not null
+#  location_id    :bigint(8)
+#  user_id        :bigint(8)
+#
+# Indexes
+#
+#  index_structures_on_location_id     (location_id)
+#  index_structures_on_structure_type  (structure_type)
+#  index_structures_on_user_id         (user_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (location_id => locations.id)
+#  fk_rails_...  (user_id => users.id)
+#
+
 class Structure < ApplicationRecord
+  ## -- RELATIONSHIPS
   belongs_to :location
   belongs_to :user, optional: true
   has_many :items, dependent: :destroy
 
-  @riddles = YAML.load_file("#{Rails.root.to_s}/config/variables/riddles.yml")
-
+  ## -- ATTRIBUTES
   enum structure_type: [:container, :wreck, :abandoned_ship, :monument]
 
+  ## — CLASS METHODS
+  def self.riddles
+    @riddles ||= YAML.load_file("#{Rails.root}/config/variables/riddles.yml")
+  end
+
+  ## — INSTANCE METHODS
   def get_items
-    Item.where(structure_id: self.id)
+    self.items
   end
 
-  # Riddles
-  def self.riddles
-    @riddles
+  def riddle
+    self.class.riddles[super]
```

Comment:
Never seen that before... How does that work?

Reply:
it calls the "parent's" ``riddle` method and then looks that value up in the class's `riddles`

Reply:
nice!